main/src/library/File.flix

```diff
@@ -119,6 +119,51 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Returns a string containing the given file `f`.
+    ///
+    pub def read(f: String): Result[String, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readString(##java.nio.file.Path);
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let result = readString(javaPath);
+            Ok(result)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns a string containing the given file `f` with the options `opts`.
+    /// TODO: add a note on what exactly the opts do
+    /// TODO: OBS I changed offset to Int32.
+    ///
+    pub def readWith(opts: {offset :: Int32, count :: Int32, charSet :: String}, f : String): Result[String, String] & Impure = {
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.FileInputStream(##java.io.File) as newInputStream;
+            import java.io.FileInputStream.read(Array[Int8], Int32, Int32);
+            import java.nio.charset.Charset:forName(String);
+            import new java.lang.String(Array[Int8], ##java.nio.charset.Charset) as newString;
+
+            let javaFile = newFile(f);
+            let stream = newInputStream(javaFile);
+            let bytes = [];
+            read(stream, bytes, opts.offset, opts.count);
```

Comment:
To not have to convert offset to Int32 here, I changed the parameter to Int32 instead of Int64. Is that okay?

Reply:
This is the right choice IMO